perses/app/cli.py

```diff
@@ -0,0 +1,159 @@
+# New cli for testing
+import datetime
+from pathlib import Path
+
+import click
+import openmm
+from openmm import OpenMMException, unit
+from openmmtools import integrators
+from openmmtools.testsystems import HarmonicOscillator
+from perses.app.setup_relative_calculation import getSetupOptions
+
+percy = """
+            _                           _
+            ;`.                       ,'/
+            |`.`-.      _____      ,-;,'|
+            |  `-.\__,-'     `-.__//'   |
+            |     `|               \ ,  |
+            `.  ```                 ,  .'
+              \_`      .     ,   ,  `_/
+                \    ^  `   ,   ^ ` /
+                 | '  |  ____  | , |
+                 |     ,'    `.    |
+                 |    (  O' O  )   |
+                 `.    \__,.__/   ,'
+                   `-._  `--'  _,'
+                       `------'
+
+                   PERSES CLI DEMO
+
+        """
+
+
+def _check_openeye_license():
+    import openeye
+
+    assert openeye.oechem.OEChemIsLicensed(), "OpenEye license checks failed!"
+
+
+def _test_gpu(platform_name):
+    from openmm.testInstallation import main
+
+    main()
+
+    # If a user asks for a platform, try and see if we can use it
+    if platform_name:
+
+        test_system = HarmonicOscillator()
+        integrator = integrators.LangevinIntegrator(
+            temperature=298.0 * unit.kelvin,
+            collision_rate=1.0 / unit.picoseconds,
+            timestep=1.0 * unit.femtoseconds,
+        )
+        try:
+            platform = openmm.Platform.getPlatformByName(platform_name)
```

Comment:
I think you want to use [`openmmtools.utils.platform_supports_precision`](https://github.com/choderalab/openmmtools/blob/main/openmmtools/utils.py#L517) instead since this will not check whether the platform supports `mixed` precision:
```python
import openmmtools
result = openmmtools.utils.platform_supports_precision('mixed')
```
This will return `True` if `mixed` precision is supported.

You could alternatively just borrow some of the code from there instead.

Reply:
Okay so that is much cleaner, but looks like I've got a bug in openmmtools to fix now:

```  
File "/home/mmh/Projects/perses/perses/app/cli.py", line 75, in _test_platform
    assert openmmtools.utils.platform_supports_precision(platform_name, 'mixed')
  File "/home/mmh/miniconda3/envs/perses/lib/python3.8/site-packages/openmmtools/utils.py", line 537, in platform_supports_precision
    platform = openmm.Platform.getPlatformByName(platform)
UnboundLocalError: local variable 'openmm' referenced before assignment
```

Reply:
Oh no! I don't understand how this didn't have test coverage. Can you be sure to add a test if you find you do indeed need to add the bugfix?

Reply:
Yes my plan is to figure out what tests are missing, add the tests, then fix the code. TDD in action!

Reply:
@ijpulidos @mikemhenry : Could this bug impact the new perses release too? Do we need to get the fix in before the new perses will work?

Reply:
Okay with some quick and dirty grep-ing, we don't use this utility in perses, and in openmmtools, the utility is also not used. I'm working on the fix now, but it looks like the test didn't ever try and use a string to set the platform name, but instead passed in an already instantiated openmm platform, so the branch here:

```
    if isinstance(platform, str):
        # Get the actual Platform object if the platform_name was specified
        platform = openmm.Platform.getPlatformByName(platform)
```
Doesn't have any test coverage.

Reply:
What am I missing here? We **are** importing openmm in the [first few lines of this code](https://github.com/choderalab/openmmtools/blob/7e6926f4719a331d06a66bc26be41d77779c3029/openmmtools/utils.py#L34-L38), aren't we? Why it cannot find the reference to openmm in this case?

Reply:
because this is weird but how python works :upside_down_face: 

```
>>> import sys
>>> def foo():
...  print(sys)
...  if True:
...   import sys
... 
>>> foo()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 2, in foo
UnboundLocalError: local variable 'sys' referenced before assignment
>>>
```

Reply:
Importing openmm down here confused python
https://github.com/choderalab/openmmtools/blob/main/openmmtools/utils.py#L547-L550

Reply:
Right, thanks. Got it!